server/src/main/scala/org/http4s/server/middleware/DefaultHead.scala

```diff
@@ -45,5 +49,10 @@ object DefaultHead {
     apply(httpRoutes)
 
   private[this] def drainBody[G[_]: Concurrent](response: Response[G]): Response[G] =
-    response.copy(body = response.body.interruptWhen[G](Stream(true)).drain)
+    response.copy(entity = response.entity match {
+      case Strict(chunk) => TrustMe(Stream.empty, chunk.size.toLong)
```

Comment:
:face_with_head_bandage:

Might `HeadBody` be a thing?  Or `DrainingBody`?  Then you've got to deal with it everywhere else, though.